File: src/go/matrix.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"time"
)

func main() {
	N := 256
	if nStr := os.Getenv("MATRIX_SIZE"); nStr != "" {
		if val, err := strconv.Atoi(nStr); err == nil {
			N = val
		}
	}

	A := make([]float64, N*N)
	B := make([]float64, N*N)
	C := make([]float64, N*N)

	for i := 0; i < N*N; i++ {
		A[i] = 1.0 + float64(i%100)*0.01
		B[i] = 1.0 - float64(i%100)*0.01
	}

	start := time.Now()

	for i := 0; i < N; i++ {
		for k := 0; k < N; k++ {
			r := A[i*N+k]
			for j := 0; j < N; j++ {
				C[i*N+j] += r * B[k*N+j]
			}
		}
	}

	elapsed := time.Since(start)

	fmt.Printf("Matrix(%dx%d)\n", N, N)
	fmt.Printf("Result[0]: %.4f\n", C[0])
	fmt.Printf("Time: %.3f ms\n", float64(elapsed.Microseconds())/1000.0)
}

File: src/rust/matrix.rs
use std::env;
use std::time::Instant;

fn main() {
    let n_str = env::var("MATRIX_SIZE").unwrap_or("256".to_string());
    let n = n_str.parse::<usize>().unwrap_or(256);

    let mut a = vec![0.0f64; n * n];
    let mut b = vec![0.0f64; n * n];
    let mut c = vec![0.0f64; n * n];

    for i in 0..(n*n) {
        a[i] = 1.0 + (i % 100) as f64 * 0.01;
        b[i] = 1.0 - (i % 100) as f64 * 0.01;
    }

    let start = Instant::now();

    // Matrix Multiplication
    for i in 0..n {
        for k in 0..n {
            let r = a[i * n + k];
            for j in 0..n {
                c[i * n + j] += r * b[k * n + j];
            }
        }
    }

    let duration = start.elapsed();

    println!("Matrix({}x{})", n, n);
    println!("Result[0]: {}", c[0]);
    println!("Time: {:.3} ms", duration.as_micros() as f64 / 1000.0);
}
